client/components/domains/register-domain-step/index.jsx

```diff
@@ -1212,6 +1212,13 @@ class RegisterDomainStep extends React.Component {
 
 	onAddDomain = ( suggestion ) => {
 		const domain = get( suggestion, 'domain_name' );
+		const { premiumDomains } = this.state;
+
+		// disable adding a domain to the cart while the premium price is still fetching
+		if ( premiumDomains?.[ domain ]?.pending ) {
+			return;
```

Comment:
I like that you doubled down here. This ensures that if the button rendering is out of whack for a reason, it doesn't get added to the cart.